lib/active_model/serializer/lazy_association.rb

```diff
@@ -62,7 +62,7 @@ def serialize_object!(object)
             #
             # Should be reflection_options[:virtual_value] or adapter needs to figure out what to do
             # with an object that is non-nil and has no defined serializer.
-            cached_result[:virtual_value] = object.try(:as_json) || object
+            cached_result[:virtual_value] = object
```

Comment:
This leads to invalid JSON:API resources, such as in https://github.com/rails-api/active_model_serializers/issues/1978